bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/ChannelPool.java

```diff
@@ -134,15 +159,51 @@ public String authority() {
   public <ReqT, RespT> ClientCall<ReqT, RespT> newCall(
       MethodDescriptor<ReqT, RespT> methodDescriptor, CallOptions callOptions) {
     Preconditions.checkState(!shutdown, "Cannot perform operations on a closed connection");
-    ClientCall<ReqT, RespT> delegate = getNextChannel().newCall(methodDescriptor, callOptions);
+    int currentRequestNum = requestCount.getAndIncrement();
+    ImmutableList<ChannelAndMetrics> channelsList = channels.get();
+    int index = Math.abs(currentRequestNum % channelsList.size());
```

Comment:
On a related note, instead of doing modulo+indexing on the list, you could use a cycling deque+re-enque pattern with an ArrayDeque, or the Guava cycling iterator: 

https://github.com/google/guava/blob/165ef0d34adffb8a5ac84b859e9d58cd68412193/guava/src/com/google/common/collect/Iterables.java#L400